pytest_django/plugin.py

```diff
@@ -40,10 +39,96 @@
+    (1, 6): ('assertContains', 'assertFieldOutput', 'assertFormError',
+             'assertFormsetError', 'assertHTMLEqual', 'assertHTMLNotEqual',
+             'assertInHTML', 'assertJSONEqual', 'assertNotContains',
+             'assertNumQueries', 'assertQuerysetEqual', 'assertRaisesMessage',
+             'assertRedirects', 'assertTemplateNotUsed', 'assertTemplateUsed',
+             'assertXMLEqual', 'assertXMLNotEqual'),
+    # from https://github.com/django/django/blob/stable/1.7.x/django/test/testcases.py#L162
+    #  and https://github.com/django/django/blob/stable/1.7.x/django/test/testcases.py#L722
+    (1, 7): ('assertContains', 'assertFieldOutput', 'assertFormError',
+             'assertFormsetError', 'assertHTMLEqual', 'assertHTMLNotEqual',
+             'assertInHTML', 'assertJSONEqual', 'assertNotContains',
+             'assertNumQueries', 'assertQuerysetEqual', 'assertRaisesMessage',
+             'assertRedirects', 'assertTemplateNotUsed', 'assertTemplateUsed',
+             'assertXMLEqual', 'assertXMLNotEqual'),
+    # from https://github.com/django/django/blob/stable/1.8.x/django/test/testcases.py#L142
+    #  and https://github.com/django/django/blob/stable/1.8.x/django/test/testcases.py#L751
+    (1, 8): ('assertContains', 'assertFieldOutput', 'assertFormError',
+             'assertFormsetError', 'assertHTMLEqual', 'assertHTMLNotEqual',
+             'assertInHTML', 'assertJSONEqual', 'assertJSONNotEqual',
+             'assertNotContains', 'assertNumQueries', 'assertQuerysetEqual',
+             'assertRaisesMessage', 'assertRedirects', 'assertTemplateNotUsed',
+             'assertTemplateUsed', 'assertXMLEqual', 'assertXMLNotEqual'),
+    # from https://github.com/django/django/blob/stable/1.9.x/django/test/testcases.py#L158
+    #  and https://github.com/django/django/blob/stable/1.9.x/django/test/testcases.py#L818
+    (1, 9): ('assertContains', 'assertFieldOutput', 'assertFormError',
+             'assertFormsetError', 'assertHTMLEqual', 'assertHTMLNotEqual',
+             'assertInHTML', 'assertJSONEqual', 'assertJSONNotEqual',
+             'assertNotContains', 'assertNumQueries', 'assertQuerysetEqual',
+             'assertRaisesMessage', 'assertRedirects', 'assertTemplateNotUsed',
+             'assertTemplateUsed', 'assertXMLEqual', 'assertXMLNotEqual'),
+    # from https://github.com/django/django/blob/stable/1.10.x/django/test/testcases.py#L155
+    #  and https://github.com/django/django/blob/stable/1.10.x/django/test/testcases.py#L797
+    (1, 10): ('assertContains', 'assertFieldOutput', 'assertFormError',
+              'assertFormsetError', 'assertHTMLEqual', 'assertHTMLNotEqual',
+              'assertInHTML', 'assertJSONEqual', 'assertJSONNotEqual',
+              'assertNotContains', 'assertNumQueries', 'assertQuerysetEqual',
+              'assertRaisesMessage', 'assertRedirects', 'assertTemplateNotUsed',
+              'assertTemplateUsed', 'assertXMLEqual', 'assertXMLNotEqual'),
+}
```

Comment:
These can be extracted dyanmically with e.g

```python
In [3]: {attr for attr in SimpleTestCase.__dict__ if attr.startswith('assert')}
Out[3]:
{'assertContains',
 'assertFieldOutput',
 'assertFormError',
 'assertFormsetError',
 'assertHTMLEqual',
 'assertHTMLNotEqual',
 'assertInHTML',
 'assertJSONEqual',
 'assertJSONNotEqual',
 'assertNotContains',
 'assertRaisesMessage',
 'assertRedirects',
 'assertTemplateNotUsed',
 'assertTemplateUsed',
 'assertXMLEqual',
 'assertXMLNotEqual'}
```

This works because a class's `__dict__` contains the names defined in it, but not looking up its MRO.

You probably need to look at `TestCase` and `TransactionTestCase` too to get them all, didn't check that.